<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<doxygen xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="compound.xsd" version="1.9.8" xml:lang="en-US">
  <compounddef id="classoneapi_1_1dal_1_1v1_1_1system__error" kind="class" language="C++" prot="public">
    <compoundname>oneapi::dal::v1::system_error</compoundname>
    <basecompoundref refid="classoneapi_1_1dal_1_1v1_1_1runtime__error" prot="public" virt="non-virtual">runtime_error</basecompoundref>
    <basecompoundref prot="public" virt="non-virtual">system_error</basecompoundref>
    <sectiondef kind="public-func">
      <memberdef kind="function" id="classoneapi_1_1dal_1_1v1_1_1system__error_1abf843cbb29dec939d0731e491bab6f70" prot="public" static="no" const="yes" explicit="no" inline="no" noexcept="yes" virt="virtual">
        <type>const char *</type>
        <definition>const char * what</definition>
        <argsstring>() const noexcept override</argsstring>
        <name>what</name>
        <qualifiedname>oneapi::dal::v1::system_error::what</qualifiedname>
        <reimplements refid="classoneapi_1_1dal_1_1v1_1_1exception_1ab146d0f106ecc0e41933be2b45a7a579">what</reimplements>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="/home/runner/work/oneDAL/oneDAL/cpp/oneapi/dal/exceptions.hpp" line="82" column="16"/>
      </memberdef>
      <memberdef kind="function" id="classoneapi_1_1dal_1_1v1_1_1system__error_1a84ca5da74d4b497e539df10bb31b14a6" prot="public" static="no" const="yes" explicit="no" inline="no" noexcept="yes" virt="non-virtual">
        <type>const std::error_code &amp;</type>
        <definition>const std::error_code &amp; code</definition>
        <argsstring>() const noexcept</argsstring>
        <name>code</name>
        <qualifiedname>oneapi::dal::v1::system_error::code</qualifiedname>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="/home/runner/work/oneDAL/oneDAL/cpp/oneapi/dal/exceptions.hpp" line="83" column="27"/>
      </memberdef>
    </sectiondef>
    <briefdescription>
    </briefdescription>
    <detaileddescription>
    </detaileddescription>
    <inheritancegraph>
      <node id="3">
        <label>exception</label>
        <link refid="classoneapi_1_1dal_1_1v1_1_1exception"/>
      </node>
      <node id="2">
        <label>runtime_error</label>
        <link refid="classoneapi_1_1dal_1_1v1_1_1runtime__error"/>
        <childnode refid="3" relation="public-inheritance">
        </childnode>
      </node>
      <node id="4">
        <label>system_error</label>
      </node>
      <node id="1">
        <label>system_error</label>
        <link refid="classoneapi_1_1dal_1_1v1_1_1system__error"/>
        <childnode refid="2" relation="public-inheritance">
        </childnode>
        <childnode refid="4" relation="public-inheritance">
        </childnode>
      </node>
    </inheritancegraph>
    <collaborationgraph>
      <node id="3">
        <label>exception</label>
        <link refid="classoneapi_1_1dal_1_1v1_1_1exception"/>
      </node>
      <node id="2">
        <label>runtime_error</label>
        <link refid="classoneapi_1_1dal_1_1v1_1_1runtime__error"/>
        <childnode refid="3" relation="public-inheritance">
        </childnode>
      </node>
      <node id="4">
        <label>system_error</label>
      </node>
      <node id="1">
        <label>system_error</label>
        <link refid="classoneapi_1_1dal_1_1v1_1_1system__error"/>
        <childnode refid="2" relation="public-inheritance">
        </childnode>
        <childnode refid="4" relation="public-inheritance">
        </childnode>
      </node>
    </collaborationgraph>
    <location file="/home/runner/work/oneDAL/oneDAL/cpp/oneapi/dal/exceptions.hpp" line="79" column="1" bodyfile="/home/runner/work/oneDAL/oneDAL/cpp/oneapi/dal/exceptions.hpp" bodystart="79" bodyend="84"/>
    <listofallmembers>
      <member refid="classoneapi_1_1dal_1_1v1_1_1system__error_1a84ca5da74d4b497e539df10bb31b14a6" prot="public" virt="non-virtual"><scope>oneapi::dal::v1::system_error</scope><name>code</name></member>
      <member refid="classoneapi_1_1dal_1_1v1_1_1system__error_1abf843cbb29dec939d0731e491bab6f70" prot="public" virt="virtual"><scope>oneapi::dal::v1::system_error</scope><name>what</name></member>
      <member refid="classoneapi_1_1dal_1_1v1_1_1exception_1a5a27ecefe1c03e5d2e35adf5032f69cf" prot="public" virt="virtual"><scope>oneapi::dal::v1::system_error</scope><name>~exception</name></member>
    </listofallmembers>
  </compounddef>
</doxygen>
